Command/Command.php

```diff
@@ -10,43 +10,32 @@
 use OpenClassrooms\Bundle\OneSkyBundle\EventListener\TranslationUploadTranslationEvent;
 use OpenClassrooms\Bundle\OneSkyBundle\Model\ExportFile;
 use OpenClassrooms\Bundle\OneSkyBundle\Model\UploadFile;
-use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Command\Command as SymfonyCommand;
 use Symfony\Component\Console\Helper\ProgressBar;
 use Symfony\Component\Console\Helper\Table;
 use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
 use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 
 const PROGRESS_BAR_FORMAT = "<comment>%message%</comment>\n %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s% %memory:6s%";
 
 /**
  * @author Romain Kuzniak <romain.kuzniak@openclassrooms.com>
  */
-abstract class Command extends ContainerAwareCommand
+abstract class Command extends SymfonyCommand
 {
-    /**
-     * @var EventDispatcherInterface
-     */
-    protected $eventDispatcher;
-
-    /**
-     * @var ProgressBar
-     */
-    private $progressBar;
-
-    /**
-     * @return string
-     */
-    abstract protected function getCommandName();
-
-    /**
-     * @return string
-     */
-    abstract protected function getCommandDescription();
+    private ProgressBar $progressBar;
+    protected ParameterBagInterface $parameters;
```

Comment:
Where does this property get set?

Reply:
Inside services.xml, I created this:
```
<parameters>
        <parameter key="openclassrooms_onesky.project_id">%openclassrooms_onesky.project_id%</parameter>
    </parameters>
```
Then, we can fetch that inside ParameterBagInterface

Reply:
But what I mean is: how does the ParameterBagInterface in the Command class (or the subclasses of it) ever get set? Maybe I overlooked it, but I don't see a setter for it, and it is not set in the constructor?

Reply:
Actually, the ParameterBagInterface is only used to get the projectId, or?

We could make the projectId a parameter in the constructor, and inject it in the service config, the same like is done [here](https://github.com/onlyfyio/OneSkyBundle/blob/e8fb4719a29e4a9436711a0b859d1a6536e1b4e9/Resources/config/services.xml#L89), but referencing the parameter something like (not sure if that syntax is quite right, or if the argument name needs specifying as well):

```
<argument type="string">%openclassrooms_onesky.project_id%</argument>
```

Reply:
Changed :)

Reply:
![](https://media.giphy.com/media/KxiRwO7tqXCTDVKobo/giphy.gif)